src/main/java/cs2103/v15_1j/jimjim/ui/TaskEventRowFactory.java

```diff
@@ -78,52 +80,51 @@ public void showAllTaskEvents(){
 					itemToAdd = nextDeadlineTask;
 					deadlineTaskCounter++;
 				}
-		    }
-		    
-		    LocalDate tempDate = getLatestDate(currentDate, itemToAdd);
-		    
-		    if(currentDate == null){
-		    	currentDate = tempDate;
-		    	addLabel(currentDate);
-		    }
-		    else if(!currentDate.equals(tempDate)){
-		    	currentDate = tempDate;
-		    	addLabel(currentDate);
-		    }
-		    
-		    addTaskEvent(itemToAdd);
+			}
+
+			LocalDate tempDate = getLatestDate(currentDate, itemToAdd);
```

Comment:
`getLatestDate` chooses the latest of either the currentDate or the nextEvent/nextDeadline Task date. I shifted the getting of the nextEvent/nextDeadlineTask date out because you had to check for alot of conditions so i made it another method.

The `tempDate` is to check whether the new date is same as the old one, if it is not then I'll create a new dateLabel